### ISSUE TITLE
npeer is maybe missing?

### ISSUE THREAD

**Issue by sbruhse** [bryanyang0528/PeerGroupAnalysis#1]
Hey!
I am looking into your code because currently I am working on a project for my BSc to analyse data and I want to use a PGA implemented in python. I read the papers of Bolton & Hand (both 2001) and of Ferdousi & Maeda (2006) and I am wondering about your solution.
How I understand the PGA you are having a fixed size for a peer group (npeer). Ferdousi & Maeda (2006) is especially explaining how they choosed npeer. By using KMeans you are getting Clusters with a various size.
Additionally both are refering to the euclidean distance, which is also used by KMeans, of course, but either are refering to kmeans directly.

Do I missunderstand something?

I hope you can help with my confusion. 

Best regards
Simon

**Comment by bryanyang0528** [comment]
Thank you for your question. There are a few approaches to choose a suitable k for k groups in KMeans like elbow point. But the `really best k` is not easy to find in this approach, it must consider the rest steps for example using the new group tag to do some ML for prediction, and that's why I didn't implement this in this project.

**Comment by sbruhse** [comment]
Thank you for your answer!
I tried hard to use one of the standard Clustering algorithms because they are so easy to implement in Python but it didnt work for my data at all. My data is mostly crowded in one big cluster and there are some global outliers.
My approach was now to build one PeerGroup for each Customer by choosing the k nearest objects with the euclidean distance. Additionally I weighted the closer peer group members higher like in https://www.sciencedirect.com/science/article/pii/S0957417412002692. That is working pretty well!
Currently I am just using a scalar value but I am also thinking about to use vectors.
